host/host.cpp

```diff
@@ -1,91 +1,130 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
-#include <assert.h>
-#include <limits.h>
-#include <openenclave/host.h>
-#include <stdio.h>
-#include <sys/stat.h>
-#include <sys/types.h>
-#include <fstream>
 #include <iostream>
-#include <iterator>
-#include <vector>
-#include "../shared.h"
-#include "redis.h"
-
-#include "ortoa_u.h"
-
-using namespace std;
-
-oe_enclave_t* enclave = NULL;
-
-bool check_simulate_opt(int* argc, const char* argv[])
-{
-    for (int i = 0; i < *argc; i++)
-    {
-        if (strcmp(argv[i], "--simulate") == 0)
-        {
-            cout << "Running in simulation mode" << endl;
-            memmove(&argv[i], &argv[i + 1], (*argc - i) * sizeof(char*));
-            (*argc)--;
-            return true;
-        }
-    }
-    return false;
+#include "../constants/network.h"
+#include "../gen-cpp/RPC.h"
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/server/TSimpleServer.h>
+#include <thrift/transport/TServerSocket.h>
+#include <thrift/transport/TBufferTransports.h>
+
+using namespace ::apache::thrift;
+using namespace ::apache::thrift::protocol;
+using namespace ::apache::thrift::transport;
+using namespace ::apache::thrift::server;
+
+class RPCHandler : virtual public RPCIf {
+ public:
+  RPCHandler() {
+    // Your initialization goes here
+  }
+
+  void access(std::string& _return, const Operation& operation) {
+    // Your implementation goes here
+    std::cout << operation.op << " " << operation.key << " " << operation.value << std::endl;
+  }
+
+};
+
+int main(int argc, char **argv) {
+  int port = HOST_PORT;
+  ::std::shared_ptr<RPCHandler> handler(new RPCHandler());
+  ::std::shared_ptr<TProcessor> processor(new RPCProcessor(handler));
```

Comment:
(question) What's going on with the namespacing here? Why is it `::std::shared_ptr` as opposed to `std::shared_ptr`?

(nit) I find it cleaner to initialize as follows:
`auto handler = std::make_shared<RPCProcessor>(handler)`

Reply:
This file generally is generated by Thrift as skeleton code for an RPC client. The namespacing is a result of the code generation. I can look into the specific reason why Thrift generates it like that

Reply:
Oh nvm then. Just looked weird